packages/run-it/src/components/ShowResponse/responseUtils.tsx

```diff
@@ -73,19 +73,73 @@ export const isColumnar = (data: any[]) => {
   return !complex
 }
 
+type ItemType = 'a' | 'o' | 'v' | 'u'
+
+/**
+ * Is this an array, an object, a value, or undefined
+ * @param value to check
+ */
+const itemType = (value: any): ItemType => {
+  if (!value) return 'u'
+  if (Array.isArray(value)) return 'a'
+  if (value instanceof Object) return 'o'
+  return 'v'
+}
+
+/**
+ * Get the 2D type mapping for the object
+ * @param json to analyze
+ */
+const getTypes = (json: any) => {
+  const types = [new Set<ItemType>(), new Set<ItemType>()]
```

Comment:
can json be a string, number, boolean or date?

Reply:
the value passed in that we _expect_ to be JSON could be any type